core/tauri/scripts/ipc-protocol.js

```diff
@@ -0,0 +1,52 @@
+// Copyright 2019-2023 Tauri Programme within The Commons Conservancy
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-License-Identifier: MIT
+
+(function () {
+  const processIpcMessage = __RAW_process_ipc_message_fn__
+  const osName = __TEMPLATE_os_name__
+  const fetchChannelDataCommand = __TEMPLATE_fetch_channel_data_command__
+  const useCustomProtocol = __TEMPLATE_use_custom_protocol__
+
+  Object.defineProperty(window, '__TAURI_POST_MESSAGE__', {
+    value: (message) => {
+      const { cmd, callback, error, payload, options } = message
+
+      // use custom protocol for IPC if the flag is set to true, the command is the fetch data command or when not on Linux/Android
+      if (useCustomProtocol || cmd == fetchChannelDataCommand || (osName !== 'linux' && osName !== 'android')) {
```

Comment:
The second comparison should probably use strict equality operator.
```suggestion
      if (useCustomProtocol || cmd === fetchChannelDataCommand || (osName !== 'linux' && osName !== 'android')) {
```